src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockSizeRule.cs

```diff
@@ -39,45 +38,14 @@ public override Task RunAsync(RuleContext context)
             Block block = context.ValidationContext.BlockToValidate;
 
             // Size limits.
-            if ((block.Transactions.Count == 0) || (block.Transactions.Count > options.MaxBlockBaseSize) ||
-                (GetSize(this.Parent.Network, block, TransactionOptions.None) > options.MaxBlockBaseSize))
+            if ((block.Transactions.Count == 0) || (block.Transactions.Count > consensus.Options.MaxBlockBaseSize) ||
```

Comment:
I don't really understand how the transaction count could ever be so large. Not that you changed the logic, but it doesn't seem to make sense unless it's possible to have a massive amount of empty transactions somehow.